dbms/src/TiDB/Schema/SchemaBuilder.cpp

```diff
@@ -936,15 +675,17 @@ void SchemaBuilder<Getter, NameMapper>::applyCreateSchema(const TiDB::DBInfoPtr
     interpreter.setForceRestoreData(false);
     interpreter.execute();
 
-    databases.emplace(KeyspaceDatabaseID{keyspace_id, db_info->id}, db_info);
+    std::unique_lock<std::shared_mutex> lock(shared_mutex_for_databases);
+    databases.emplace(db_info->id, db_info);
```

Comment:
```suggestion
    {
        std::unique_lock<std::shared_mutex> lock(shared_mutex_for_databases);
        databases.emplace(db_info->id, db_info);
    }
```
We can limit the lock scope so that logging won't hold the lock